### ISSUE TITLE
Include UBC as the spawn realm

### ISSUE THREAD

**Issue by lukasgolson** [ubc-minetest-classroom/Minetest_Classroom#65]
With PR #64, it's now possible to support realms of any size.

I'll configure the UBC schematic(s) to be placed in a spawnRealm.

**Comment by pauldpickell** [comment]
Assets are [here](https://github.com/pauldpickell/UBC_Campus_Minetest_Map).

Edit: Just a note that when spawn realm schematic selection and/or mapgen options are introduced on initiation of the world, we need to update the "Go Home" [button](https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/df4c0c806e8199333bbca5c3f435f587eb8411b6/mods/mc_student/init.lua#L286-L297):
```
                if fields.spawn then
                        -- TODO: dynamically extract the static spawn point from the minetest.conf file
                        -- local cmeta = Settings(minetest.get_modpath("mc_teacher").."/maps/"..map..".conf")
                        -- local spawn_pos_x = tonumber(cmeta:get("spawn_pos_x"))
                        -- local spawn_pos_y = tonumber(cmeta:get("spawn_pos_y"))
                        -- local spawn_pos_z = tonumber(cmeta:get("spawn_pos_z"))
                        local spawn_pos = {
                                x = 1426,
                                y = 92,
                                z = 1083,
                        }
                        player:set_pos(spawn_pos)
```

**Comment by pauldpickell** [comment]
Hmm, not sure if the UBC schematic sequence is the best example to develop this feature from. Ideally we would have a single schematic to work with. I wonder if it's possible to import another world using the `map.sqlite` file?

**Comment by pauldpickell** [comment]
A possible work around for now might just be to provide the option to set the spawn location for the spawn realm (#67) and define antipodes `pos1` and `pos2` in-game (similar to `Minetest-WorldEdit`) to get the dimensions in order to block off that coordinate space in the realm system. All this without invoking map barriers or the ability to export a schematic. The UBC campus is already built in a singlenode world and generated at `{x=0, y=0, z=0}` so there should be no coordinate conflicts with other realms. 

I cannot see the need for feature support beyond this as it is very unlikely that anyone would come along with massively large pre-built maps. The only other solution I can see is fully integrating the LiDAR-to-nodes conversion process (i.e., [`gis2minetest`](https://github.com/pauldpickell/gis2minetest)) as an option for generating a realm, which would be so cool, but also probably beyond our scope. This is the only way that we could fully control the dimensions, placement, timing, and chunking of large areas.

**Comment by lukasgolson** [comment]
> A possible work around for now might just be to provide the option to set the spawn location for the spawn realm (#67) and define antipodes `pos1` and `pos2` in-game (similar to `Minetest-WorldEdit`) to get the dimensions in order to block off that coordinate space in the realm system. All this without invoking map barriers or the ability to export a schematic. The UBC campus is already built in a singlenode world, so



> A possible work around for now might just be to provide the option to set the spawn location for the spawn realm (#67) and define antipodes `pos1` and `pos2` in-game (similar to `Minetest-WorldEdit`) to get the dimensions in order to block off that coordinate space in the realm system. All this without invoking map barriers or the ability to export a schematic. The UBC campus is already built in a singlenode world and generated at `{x=0, y=0, z=0}` so there should be no coordinate conflicts with other realms.
> 
> I cannot see the need for feature support beyond this as it is very unlikely that anyone would come along with massively large pre-built maps. The only other solution I can see is fully integrating the LiDAR-to-nodes conversion process (i.e., [`gis2minetest`](https://github.com/pauldpickell/gis2minetest)) as an option for generating a realm, which would be so cool, but also probably beyond our scope. This is the only way that we could fully control the dimensions, placement, timing, and chunking of large areas.

Very good points; I've just finished adding in a command to manually define a realm from pos1 to pos2 as well as a command to set a realm as the spawn realm.

That said, the integration between the UBC map is now complete (and didn't take as much time as I thought it would). As a bonus, it serves as a good example for injecting custom logic into a realm via configuration files and simple scripts. I'll submit a PR to the UBC map repo for the optional integration and I'll push the new commands. 

EDIT: After thinking about this, I realized that defining a realm using the command does not exclude those chunks from our placement logic. Adding arbitrary "excluded" chunks might be a bit tough with how the system is currently set-up (and reduce performance if we're not careful); but I'll look into it further before saying anything concrete.

**Comment by lukasgolson** [comment]
Solved in #81; See https://github.com/ubc-minetest-classroom/Minetest_Classroom/wiki/How-to-import-the-UBC-map-as-spawn for instructions for using this feature.
